tracer/test/Datadog.Trace.Tests/SpanContextPropagatorTests.cs

```diff
@@ -3,34 +3,254 @@
+            var headers = new Mock<IHeadersCollection>();
+
+            SpanContextPropagator.Instance.Inject(context, headers.Object);
+
+            VerifySetCalls(headers);
+        }
+
+        [Fact]
+        public void Inject_CarrierAndDelegate()
+        {
+            var context = new SpanContext(TraceId, SpanId, SamplingPriority, serviceName: null, Origin);
+            var headers = new Mock<IHeadersCollection>();
+
+            SpanContextPropagator.Instance.Inject(context, headers.Object, (carrier, name, value) => carrier.Set(name, value));
+
+            VerifySetCalls(headers);
+        }
+
+        [Fact]
+        public void Extract_IHeadersCollection()
+        {
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            var headers = CreatePopulatedHeaders();
+            object result = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_CarrierAndDelegate()
+        {
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            var headers = CreatePopulatedHeaders();
+            object result = SpanContextPropagator.Instance.Extract(headers.Object, (carrier, name) => carrier.GetValues(name));
+
+            VerifyGetCalls(headers);
+
+            result.Should()
+                  .BeEquivalentTo(
+                       new
+                       {
+                           TraceId,
+                           SpanId,
+                           Origin,
+                           SamplingPriority
+                       });
+        }
+
+        [Fact]
+        public void Extract_EmptyHeadersReturnsNull()
+        {
+            var headers = new Mock<IHeadersCollection>();
+
+            // use `object` so Should() below works correctly,
+            // otherwise it picks up the IEnumerable<KeyValuePair<string, string>> overload
+            object resultContext = SpanContextPropagator.Instance.Extract(headers.Object);
+
+            resultContext.Should().BeNull();
+        }
+
         [Fact]
         public void SpanContextRoundTrip()
         {
-            const ulong expectedTraceId = 1;
-            const ulong expectedSpanId = 2;
-            const SamplingPriority expectedSamplingPriority = SamplingPriority.UserKeep;
-            const string expectedOrigin = "origin";
+            var context = new SpanContext(TraceId, SpanId, SamplingPriority, serviceName: null, Origin);
```

Comment:
The tests above this `SpanContextRoundTrip()` are now. `SpanContextRoundTrip()` was already here, and the rest of the tests below were moved here from `HeadersCollectionTestHelpers.cs`. All have been refactored use Moq, FluentAssertions, etc.